src/groupByKey.ts

```diff
@@ -1,7 +1,22 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
+type GroupsMap<Group> = {
+  [key: string]: Group[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Group>(
+  items: Group[],
+  key: keyof Group,
+): GroupsMap<Group> {
+  const grouppedItems: GroupsMap<Group> = items.reduce((result, current) => {
+    const itemKey = current[key] as string;
+
+    if (result[itemKey]) {
+      result[itemKey] = [...result[itemKey], current];
+    } else {
+      result[itemKey] = [current];
+    }
+
+    return result;
+  }, {} as GroupsMap<Group>);
```

Comment:
Using the spread operator inside the loop can lead to performance issues because it creates a new array every time it is executed. Instead, you can directly push the current item to the existing array. This can be done by changing line 13 to `result[itemKey].push(current);`.